app.jsx

```diff
@@ -11,69 +11,75 @@ import styled from "styled-components";
 
 import Colors from "./constants/colors";
 import Spacing from "./constants/spacing";
+import { BASE_STYLES, TAG_STYLES } from "./constants/styles";
 
-const parseNodes = (nodes, baseStyle = "normal") => {
+const getInlineStyle = (inlineStyle) => {
+    const isItalic = !!inlineStyle.match(/italic/);
+    const isBold = !!inlineStyle.match(/weight: 600|700|bold/);
+
+    let style = [];
+    if (isItalic) {
+        style = [...style, BASE_STYLES.ITALIC];
+    }
+    if (isBold) {
+        style = [...style, BASE_STYLES.BOLD];
+    }
+    
+    return style.length > 0 ? style : [];
+}
+
+const getStyle = ({name, attribs}) => {
+    const tagStyles = (name && TAG_STYLES[name] && [TAG_STYLES[name]]) || [];
+    const inlineStyles = (attribs && attribs.style && getInlineStyle(attribs.style)) || [];
+    const aggregatedStyles = [...tagStyles, ...inlineStyles];
+    return aggregatedStyles.length > 0 ? aggregatedStyles : [BASE_STYLES.NORMAL];
+}
+
+const parseNodes = (nodes, el) => {
     let parsed = [];
     for (const node of nodes) {
-        const { attribs, children, data, name } = node;
-        if (!name) {
+        const { children, data, parent } = node;
+
+        let elementStyle;
```

Comment:
3. The second aspect we want to deal with is passing the style of each string. I realised that some elements hold several styles at the same time (for example, something can be bold, italic, and a list), so rather than creating strings that would hold all of these parameters (bold-italic-list), I decided to make elementStyle a list that holds all styles. I realise this depends on what we want to do with the final implementation, but not knowing about it I find this easier to work with. Rather than using an array, elementStyle is a Set. Indeed by adding style elements recursively through diving in nested html elements, we sometimes end up with the same style added several times to the array. Set only accepts new elements, so this means the end result will only show a list with unique elements.